### DISCUSSION TITLE
ESlint v9

### DISCUSSION THREAD

**Discussion by HenrikFricke** [remix-run/remix#9288] [Q&A] [answered]
ESlint recently updated to v9. With v9, ESlint introduced a new flat config file (`eslint.config.js`) and therefore we need to update the way we import plugins/extends. I tried updating the ESlint file, but then I wasn't sure how to use the `@remix-run/eslint-config` with the newest config file. 

Would it be better to wait until Remix supports ESlint v9?

**Comment by adamzerner** [comment]
Note: The [docs for `@remix-run/eslint-config`](https://www.npmjs.com/package/@remix-run/eslint-config) are incorrect. They say:

> This package requires at least version 8.1 of ESLint

which implies that using v9 of ESLint is compatible with `@remix-run/eslint-config`. However, `@remix-run/eslint-config` [peer depends](https://github.com/remix-run/remix/blob/main/packages/remix-eslint-config/package.json#L51) on ESLint v8.

**Reply by sergiodxa** [reply, marked as answer]
The `@remix-run/eslint-config` package is deprecated, new app doesn't come with it included anymore, as you can see in the [`.eslintrc.cjs`](https://github.com/remix-run/remix/blob/7e271a4c2c7f7766944ee4fe07d615ff669ee7ce/templates/remix/.eslintrc.cjs)

**Reply by adamzerner** [reply]
I see.

The docs should be updated to reflect the fact that the package is deprecated. What would be the best way to file a request for that?

Are there any guidelines for migrating off of `@remix-run/eslint-config`? The app I'm currently working on is about two years old and was created using the [Blues Stack template](https://github.com/remix-run/blues-stack/tree/main).

**Reply by sergiodxa** [reply]
You could send a PR to update the readme, or open an issue.

If you didn't customized your ESLint, you could copy the new default config.

**Reply by adamzerner** [reply]
> You could send a PR to update the readme

I assume it'd be desirable to add context in addition to saying that it's been deprecated and I am not knowledgeable enough to provide that context.

> or open an issue.

The [New Issue page](https://github.com/remix-run/remix/issues/new/choose) looks like it only accepts bug reports, and this isn't really a bug.

> If you didn't customized your ESLint, you could copy the new default config.

Ah, gotcha.

**Reply by shawnmclean** [reply]
If I didn't come across this discussion, how would one know this was deprecated? Are these discussions being held outside of github?

**Reply by kiliman** [reply]
Here's the issue and associated PR. It doesn't look like it's been called out in the docs or CHANGELOG, though.

https://github.com/remix-run/remix/issues/7056
https://github.com/remix-run/remix/pull/7597
